src/library/dao/playlistdao.cpp

```diff
@@ -380,15 +380,60 @@ bool PlaylistDAO::isHidden(const int playlistId) const {
     return true;
 }
 
-void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const int position)
-{
+void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const TrackId& trackId) {
+    ScopedTransaction transaction(m_database);
+
+    QSqlQuery query(m_database);
+    query.prepare("SELECT position FROM PlaylistTracks WHERE playlist_id=:id "
+                "AND track_id=:track_id");
+    query.bindValue(":id", playlistId);
+    query.bindValue(":track_id", trackId.toVariant());
+
+    query.setForwardOnly(true);
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return;
+    }
+
+    while (query.next()) {
+    	int position = query.value(query.record().indexOf("position")).toInt();
+        removeTracksFromPlaylistInner(playlistId, position);
+
+    }
+
+    transaction.commit();
+    emit(changed(playlistId));
+}
+
+
+void PlaylistDAO::removeTrackFromPlaylist(const int playlistId, const int position) {
     // qDebug() << "PlaylistDAO::removeTrackFromPlaylist"
     //          << QThread::currentThread() << m_database.connectionName();
     ScopedTransaction transaction(m_database);
+    removeTracksFromPlaylistInner(playlistId, position);
+    transaction.commit();
+    emit(changed(playlistId));
+}
+
+void PlaylistDAO::removeTracksFromPlaylist(const int playlistId, QList<int>& positions) {
+    // get positions in reversed order
+    qSort(positions.begin(), positions.end(), qGreater<int>());
+
+    //qDebug() << "PlaylistDAO::removeTrackFromPlaylist"
+    //         << QThread::currentThread() << m_database.connectionName();
+    ScopedTransaction transaction(m_database);
     QSqlQuery query(m_database);
+    foreach (int position , positions) {
+    	removeTracksFromPlaylistInner(playlistId, position);
+    }
+    transaction.commit();
+    emit(changed(playlistId));
+}
 
+void PlaylistDAO::removeTracksFromPlaylistInner(int playlistId, int position) {
+    QSqlQuery query(m_database);
     query.prepare("SELECT track_id FROM PlaylistTracks WHERE playlist_id=:id "
-                  "AND position=:position");
+                "AND position=:position");
```

Comment:
BTW, this is the correct formatting for a hanging indent per the Google C++ Style Guide. What you've changed it to is only correct in this case:

```c++
query.prepare(
        "....");
```